### ISSUE TITLE
my Error it still run but not working!

### ISSUE THREAD

**Issue by Ahmet-pc** [2008Choco/VeinMiner#94]
[20:41:06 ERROR]: Could not pass event BlockBreakEvent to VeinMiner v2.0.0
java.lang.IllegalArgumentException: Unknown GameMode with id "SURVIVAL"
        at wtf.choco.veinminer.platform.GameMode.getByIdOrThrow(GameMode.java:82) ~[?:?]
        at wtf.choco.veinminer.listener.BreakBlockListener.onBlockBreak(BreakBlockListener.java:113) ~[?:?]
        at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:37) ~[patched_1.17.1.jar:git-Paper-411]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.1.jar:git-Paper-411]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.1.jar:git-Paper-411]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.server.level.ServerPlayerGameMode.destroyBlock(ServerPlayerGameMode.java:402) ~[?:?]
        at net.minecraft.server.level.ServerPlayerGameMode.destroyAndAck(ServerPlayerGameMode.java:356) ~[?:?]
        at net.minecraft.server.level.ServerPlayerGameMode.handleBlockBreakAction(ServerPlayerGameMode.java:316) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handlePlayerAction(ServerGamePacketListenerImpl.java:1739) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.handle(ServerboundPlayerActionPacket.java:34) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundPlayerActionPacket.handle(ServerboundPlayerActionPacket.java:8) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1426) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:192) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1404) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1397) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:110) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1536) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1282) ~[patched_1.17.1.jar:git-Paper-411]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by 2008Choco** [comment]
Not certain why this happens.

For users seeing this same issue later, this user came into Discord shortly after this issue was created and I helped them there. The solution is just to downgrade to VeinMiner 1.17.10, the version before 2.0.0 was released, which works on Minecraft 1.17.1 for sure. It's possible that I didn't properly assess the supported versions for 2.0.0, this version of Paper had a bad commit, or it was a modded version of Paper.

**Comment by 2008Choco** [comment]
Reopening this issue because another user approached me with a similar error on 1.19.4
```
[12:22:56 ERROR]: Could not pass event BlockBreakEvent to VeinMiner v2.0.0
java.lang.IllegalArgumentException: Unknown GameMode with id "SURVIVAL"
	at wtf.choco.veinminer.platform.GameMode.getByIdOrThrow(GameMode.java:82) ~[?:?]
	at wtf.choco.veinminer.listener.BreakBlockListener.onBlockBreak(BreakBlockListener.java:113) ~[?:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:git-Paper-550]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.19.4.jar:git-Paper-550]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.19.4.jar:git-Paper-550]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at xyz.xenondevs.nova.util.EventUtilsKt.callEvent(EventUtils.kt:47) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.world.block.logic.break.BlockBreaker.breakBlock(BlockBreaker.kt:240) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.world.block.logic.break.BlockBreaking.handleDestroyStop(BlockBreaking.kt:164) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.world.block.logic.break.BlockBreaking.access$handleDestroyStop(BlockBreaking.kt:41) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.world.block.logic.break.BlockBreaking$handlePlayerAction$3.invoke(BlockBreaking.kt:188) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.world.block.logic.break.BlockBreaking$handlePlayerAction$3.invoke(BlockBreaking.kt:188) ~[Nova-0.13.jar:?]
	at xyz.xenondevs.nova.util.SchedulerUtilsKt.runTask$lambda$1(SchedulerUtils.kt:25) ~[Nova-0.13.jar:?]
	at org.bukkit.craftbukkit.v1_19_R3.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.4.jar:git-Paper-550]
	at org.bukkit.craftbukkit.v1_19_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[paper-1.19.4.jar:git-Paper-550]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1483) ~[paper-1.19.4.jar:git-Paper-550]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.19.4.jar:git-Paper-550]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1397) ~[paper-1.19.4.jar:git-Paper-550]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1174) ~[paper-1.19.4.jar:git-Paper-550]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.19.4.jar:git-Paper-550]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```
I will have to investigate this because clearly it's some sort of incompatibility with Paper or something...
